internal/webhooks/test/clusterclass_test.go

```diff
@@ -0,0 +1,421 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	utilfeature "k8s.io/component-base/featuregate/testing"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/feature"
+	"sigs.k8s.io/cluster-api/internal/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TestClusterClassWebhook_Succeed_Create tests the correct creation behaviour for a valid ClusterClass.
+func TestClusterClassWebhook_Succeed_Create(t *testing.T) {
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+	g := NewWithT(t)
+
+	ns, err := env.CreateNamespace(ctx, "test-topology-cluster-reconcile")
```

Comment:
```suggestion
	ns, err := env.CreateNamespace(ctx, "test-clusterclass-webhook")
```
or something other unique :)

Reply:
same below. I guess they don't have to be different between funcs in the same package as those tests are always run sequentially? (while different packages can be tested in parallel)

Reply:
Still should be changed - oversight from me.